### ISSUE TITLE
[AppImage] System theme does not work; dark theme contains white space

### ISSUE THREAD

**Issue by simoniz0r** [mujx/nheko#137]
### System:

- Nheko commit/version: Latest AppImage release as of 11-25-2017
- Operating System: Xubuntu 16.04 and Ubuntu 18.04 daily with LXQt
- Qt version: 5.5.1 on 16.04? (not sure which version) and 5.9.2 on 18.04
- C++ compiler: N/A
- Desktop Environment: Xfce and LXQt

### Actual behavior
System theme does nothing.  Dark theme contains white space below the channel list
### Expected behavior
System theme should follow the colors of my system.  Dark theme should not have white space bellow the channel list
### Steps to reproduce
Try to use the system theme and dark theme in the AppImage release.

Also, not sure if this is a bug, but nheko must be restarted before the theme will change.

Screenshot of the dark theme with white space under the channel list:

![image](https://user-images.githubusercontent.com/16279904/33233516-9efa5768-d1dc-11e7-9b63-07b40ee9f4fc.png)

Screenshot of the system theme not following system colors:

![image](https://user-images.githubusercontent.com/16279904/33233539-00efa73e-d1dd-11e7-8e77-751e622c643c.png)

**Comment by mujx** [comment]
It might be an issue with the theme you're using if the system option has no effect. I will investigate some more.

**Comment by simoniz0r** [comment]
I'm using KDE's release of Breeze dark for GTK on both systems, but using other themes doesn't seem to work either.  I just tried a half dozen different themes with no effect.  Other Qt based apps are themed fine on both systems.

**Comment by Ralith** [comment]
This is not AppImage specific. Discussion with @vurpo suggests that necessary changes from the system theme PR were not merged.

**Comment by simoniz0r** [comment]
Seems like there's a discusson going on over here about this... might be some useful info for you there.  probonopd/linuxdeployqt/issues/60

**Comment by simoniz0r** [comment]
I just wanted to confirm that this is actually only an issue with the AppImage release.  I was able to get the rpm build from Fedora's copr to install with a little extra work, and the system theme option does work when installed using the rpm.

![image](https://user-images.githubusercontent.com/16279904/33793570-04c37b2c-dc7f-11e7-84d1-502a3774ddf4.png)

**Comment by mujx** [comment]
@simoniz0r Good to know. What do you mean by 'extra work'? Do you have any more info so we can adjust the AppImage?

**Comment by simoniz0r** [comment]
It was just a dependency issue because I was using an rpm that was built for Fedora.  I had to make a symlink for a lib that was named differently on OpenSUSE than it was on Fedora.  Nothing to do with the AppImage build AFAIK.

**Comment by simoniz0r** [comment]
I just installed the RPM build from the releases page here, and the system theme is not working with the RPM release either.  This is kind of strange as it was working with the RPM release that was made for [Fedora from Fedora's copr](https://copr-be.cloud.fedoraproject.org/results/xvitaly/matrix/).

![image](https://user-images.githubusercontent.com/16279904/34361589-ff790c62-ea30-11e7-91af-bb15d6937558.png)

**Comment by mujx** [comment]
The rpm on github releases just uses the AppImage (it has no dependencies), so it's still a problem with the AppImage. 

I wonder if building the AppImage on the target system will fix the issue. Could you run the `.ci/linux/deploy.sh` script that builds the AppImage on your system?

**Comment by simoniz0r** [comment]
> Could you run the .ci/linux/deploy.sh script that builds the AppImage on your system?

Sorry, I don't have the necessary packages for building nheko or the AppImage of nheko installed on this system.  Installing qmake is another 141 packages for me lol

**Comment by Swedneck** [comment]
I installed nheko-git from the AUR on manjaro running gnome, system theme doesn't work for me either.

**Comment by PureTryOut** [comment]
@simoniz0r note that for me Breeze Dark (set as both Qt and GTK theme) works right now with the latest AppImage. Could you try the nightly again?

**Comment by simoniz0r** [comment]
Nope, no dice.  nheko is either missing some deps for this to work or it needs to be compiled on a different platform for this to work.

**Comment by mujx** [comment]
@simoniz0r Can you point us to an AppImage that works with the system's theme? Seems weird that works for @PureTryOut, maybe he has some extra libraries installed?

**Comment by simoniz0r** [comment]
The only one I can think of that follows my system theme that doesn't provide an option to use the GTK theme would be [the VLC AppImage](https://github.com/darealshinji/vlc-AppImage/releases/tag/2.2.8).

**Comment by unclechu** [comment]
I can say it was working before, nheko was colored exactly as my system theme, I have configured both QT4 and QT5 for using GTK theme (which is Adwaita-dark) or Adwaita-dark for QT5 (because in qt5ct there's no option for using GTK theme) but it looks like nheko ignores it.

**Comment by unclechu** [comment]
I also have had correctly colored file dialog with system theme, it was all dark, as it supposed to be with Adwaita-dark, but with "Dark" theme of nheko gets me something strange (@mujx is it supposed to be reported as separated issue?):
![screenshot from 2018-05-04 16-33-10](https://user-images.githubusercontent.com/799353/39625856-dacfd59c-4fb8-11e8-9bdd-ec72b97dc6f7.png)

**Comment by simoniz0r** [comment]
> because in qt5ct there's no option for using GTK theme

offtopic, but there is; it's the GTK Qt Platformtheme plugin.  The name differs quite a bit from distro to distro.

**Comment by unclechu** [comment]
@simoniz0r Thanks for notice but I can't find anything for QT5 in Fedora 28 that would looks like this. Anyway, choosing Adwaita-dark theme works well for most applications (except nheko but used to work few releases before).

**Comment by unclechu** [comment]
**UPDATE** I realized in my case system theme wasn't working because I've been using nheko RPM package from github which is (as a DEB one) just a wrapper for AppImage one, and AppImage encapsulates stuff and doesn't use system theme. So now in Fedora 28 Workstation I installed nheko 0.4.0 from fedora-updates repo and everything is working properly now, system colors, correct file dialog appearance (as anywhere else in my system) and custom font family I set in my qt config.

**Comment by unclechu** [comment]
@simoniz0r I see it probably related to your case too, because you too use AppImage. I'm not very familiar with AppImage but it may be not the nheko's issue.

**Comment by simoniz0r** [comment]
Yeah, the rpm from Fedora's copr works fine for me with the system theme, but there's a lib or two that don't match up, so I have to make symlinks for nheko to work.

When I get some time, I'll try to steal the spec file from the copr build and see if I can't move it over to the openSUSE Build Service and set it up for openSUSE deps.
